src/autopas/cells/ReferenceParticleCell.h

```diff
@@ -235,7 +263,7 @@ class ReferenceParticleCell : public ParticleCell<Particle> {
   SoA<SoAArraysType> _particleSoABuffer;
 
  private:
-  AutoPasLock _particlesLock;
+  mutable AutoPasLock _particlesLock;
```

Comment:
why do we even have this lock? Shouldn't we use `_cellLock` from the parent class?

Reply:
Good point, I came across this as well while writing the new functions and wanted to ask you if there was any reason why this extra lock was used. But then I forgot to ask.
Seems also to work with the lock from parent... So done in [06385c2](https://github.com/AutoPas/AutoPas/pull/782/commits/06385c264cc416ebc407bb395db2e8d1a70c1cd0)